### PR TITLE
[SMALLFIX] Improvements to shell scripts.

### PR DESCRIPTION
**Pull request by jsimsa**
This PR makes the following improvements to shell scripts:

- `bin/alluxio-start.sh proxies` is now supported
- `bin/alluxio-stop.sh local` is now supported
- `bin/alluxio-stop.sh proxies` is now supported

- `bin/alluxio-start.sh all` now correctly starts proxies
- `bin/alluxio-stop.sh *` stops services in the reverse order the corresponding `start` action started them

- removes brittle action type detection from `task.log`
- the `task.log` output lines of remote ssh execution are now tagged with hostname

### CODE REVIEW

File: bin/alluxio-start.sh [hunk 1]
```diff
@@ -21,15 +21,17 @@ BIN=$(cd "$( dirname "$0" )"; pwd)
 
 USAGE="Usage: alluxio-start.sh [-hNw] ACTION [MOPT] [-f]
 Where ACTION is one of:
-  all [MOPT]     \tStart master, proxy and all workers.
-  local [MOPT]   \tStart a master and worker locally.
+  all [MOPT]     \tStart master and all proxies and workers.
+  local [MOPT]   \tStart a master, proxy and worker locally.
```
Comment (calvinjia) [inline bin/alluxio-start.sh]: `proxy,`

File: bin/alluxio-stop.sh [hunk 2]
```diff
@@ -19,51 +19,64 @@ BIN=$(cd "$( dirname "$0" )"; pwd)
...
 Where component is one of:
-  all     \tStop master, proxy and all workers.
+  all     \tStop master and all proxies and workers.
+  local   \tStop local master, proxy, and worker.
   master  \tStop local master.
   proxy   \tStop local proxy.
+  proxies \tStop all remote proxies.
   worker  \tStop local worker.
-  workers \tStop local worker and all remote workers.
+  workers \tStop all remote workers.
```
Comment (calvinjia) [inline bin/alluxio-stop.sh]: Has this been changed to only `remote` if localhost is in the workers file?

File: bin/alluxio-start.sh (hunk 1 above)
Comment (jsimsa) [inline bin/alluxio-start.sh]: done

File: bin/alluxio-stop.sh (hunk 2 above)
Comment (jsimsa) [inline bin/alluxio-stop.sh]: this has been changed only to remote workers for symmetry with `start workers`

**Comment by jsimsa** [comment]
@calvinjia thanks for your review, PTAL

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**